packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx

```diff
@@ -0,0 +1,196 @@
+import {
+  buildSearchStatus,
+  SearchStatus,
+  SearchStatusState,
+} from '@coveo/headless';
+import {Component, h, Listen, State, Element} from '@stencil/core';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {Button} from '../../../common/button';
+import {Hidden} from '../../../common/hidden';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+import ArrowBottomIcon from 'coveo-styleguide/resources/icons/svg/arrow-bottom-rounded.svg';
+import {
+  InitPopoverEventPayload,
+  ClearPopoversEventPayload,
+} from './popover-event';
+
+/**
+ * @internal
+ * The `atomic-popover` component displays any facet as a popover menu.
+ * @part popover-button - The button to click to display or hide the popover menu.
+ * @part label - The popover button label.
+ * @part arrow-icon - The arrow icon on the button to display or hide the popover menu.
+ * @part placeholder - The placeholder displayed when the facet is loading.
+ * @part popover-wrapper - The wrapper that contains the 'popover-button' and the 'slot-wrapper'.
+ * @part slot-wrapper - The wrapper that contains the 'facet' or 'slot'.
+ */
+
+@Component({
+  tag: 'atomic-popover',
+  styleUrl: 'atomic-popover.pcss',
+  shadow: true,
+})
+export class AtomicPopover implements InitializableComponent {
+  @InitializeBindings()
+  public bindings!: Bindings;
+  public searchStatus!: SearchStatus;
+  @BindStateToController('searchStatus')
+  @State()
+  public searchStatusState!: SearchStatusState;
+  @State()
+  public error!: Error;
+
+  @State() public isMenuVisible = false;
+  @State() public facetId?: string;
+  @State() public facetLabel?: string = 'no-label';
+  @Element() host!: HTMLElement;
+  private facetElement?: HTMLElement;
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+
+    if (this.host.children.length === 0) {
+      this.error = {
+        name: 'No child in popover',
+        message: 'One child is required inside a set of popover tags',
+      };
+    } else if (this.host.children.length > 1) {
+      this.error = {
+        name: 'Too many children in popover',
+        message: 'Cannot have more than one child inside a set of popover tags',
+      };
+    }
```

Comment:
we generally prefer early returns rather than else ifs
```suggestion
    if (this.host.children.length === 0) {
      this.error = {
        name: 'No child in popover',
        message: 'One child is required inside a set of popover tags',
      };
      return;
    } 
    
    if (this.host.children.length > 1) {
      this.error = {
        name: 'Too many children in popover',
        message: 'Cannot have more than one child inside a set of popover tags',
      };
    }
```